dbt-materialize: enable cluster use, test suite profiles (#14195)
MaterializeInc/materialize: misc/dbt-materialize/CHANGELOG.md

# dbt-materialize Changelog
## 1.1.4 - 2022-08-24

> * Enable cluster use for materialize cloud deployments.

Reviewer: Recommend a bit more detail here!

```suggestion
* Enable configuration of [clusters](https://materialize.com/docs/unstable/sql/create-cluster/#conceptual-framework), which are a feature in a forthcoming version of Materialize Cloud, via:

  * A new `cluster` connection parameter, which specifies the default cluster for the connection.
  * A new `materialize_cluster` option for materializedview materializations, which specifies the cluster for an individual materialized view.
```

  ```sql
  {{ config(materialize_cluster='not_default', materialized='materializedview') }}
    SELECT ...
  ```

> * Run tests against materialize/materialized:v0.26.4 and cloud(main).

Reviewer: I'd strike this! The policy that we follow for release notes in the main repository is to only talk about things that have actual impact on end users.

```suggestion
```

# dbt-materialize Changelog
## 1.1.4 - 2022-08-24

* Enable configuration of [clusters](https://materialize.com/docs/unstable/sql/create-cluster/#conceptual-framework), which are a feature in a forthcoming version of Materialize Cloud, via:

  * A new `cluster` connection parameter, which specifies the default cluster for the connection.

>   * A new `materialize_cluster` option for materializedview materializations, which specifies the cluster for an individual materialized view.

Reviewer: ```suggestion
  * A new `cluster` option for materializedview materializations, which specifies the cluster for an individual materialized view.
```

# dbt-materialize Changelog
## 1.2.0 - 2022-08-31

* Enable additional configuration for indexes created on view,
  materializedview, or source materializations. Fix to use Materialize's
  internal naming convention when creating indexes without providing
  explicit names.

  * A new optional `name` parameter:
  ```sql
  {{ config(materialized='materializedview',
  indexes=[{'columns': ['col_1'], 'name':'col_1_idx'}]) }}
  SELECT ...
  ```

>   * A new `default` parameter. Default to false. If set to True, will create

Reviewer: ```suggestion
  * A new `default` parameter. Defaults to `False`. If set to `True`, will create
```

    default primary indexes.
  ```sql
  {{ config(materialized='materializedview',
    indexes=[{'default': True}]) }}
    SELECT ...
  ```

* Enable configuration of [clusters](https://materialize.com/docs/unstable/sql/create-cluster/#conceptual-framework),
  which are a feature in a forthcoming version of Materialize Cloud, via:

  * A new `cluster` connection parameter, which specifies the default cluster
    for the connection.
  * A new `cluster` option for materializedview and view materializations,

>     which specifies the cluster for an individual [materialized] view.

Reviewer: I think the `[materialized]` here doesn't link anywhere!

Reviewer: Also, does this option have any effect on view materializations? Is it used as the default for any indexes created on that view?

Reviewer: Ah, looks like yes!

Author: My bad, I meant the optional [materialized] instead of a link. Updated!

>   * A new `cluster` option for materializedview and view materializations,
>     which specifies the cluster for an individual [materialized] view.

Reviewer: ```suggestion
  * A new `cluster` option for materializedview and view materializations.
     For materialized views, this determines the cluster in which the materialized view
     is created. For both views and materializedviews, this also determines the cluster
     in which any indexes are created by default. If unspecified, the default cluster for
     the connection is used.
```
